### PR TITLE
chore: migrate codegen to controller events

### PR DESCRIPTION
**Pull request by pavelfeldman**

### CODE REVIEW

File: packages/recorder/src/callLog.tsx [hunk 1]
```diff
@@ -54,7 +54,7 @@ export const CallLogView: React.FC<CallLogProps> = ({
             { message.trim() }
           </div>;
         })}
-        { !!callLog.error && <div className='call-log-message error' hidden={!isExpanded}>{ callLog.error.error?.message }</div> }
+        { !!callLog.error && <div className='call-log-message error' hidden={!isExpanded}>{ callLog.error }</div> }
```
Comment (dgozman) [inline packages/recorder/src/callLog.tsx:57]: Stray change?

File: packages/playwright-core/src/server/reuseController.ts [hunk 2]
```diff
@@ -61,9 +62,9 @@ export class ReuseController extends SdkObject {
         onPageNavigated: () => this._emitSnapshot(),
         onPageClose: () => this._emitSnapshot(),
       };
-      this.instrumentation.addListener(this._trackHierarchyListener, null);
+      this._playwright.instrumentation.addListener(this._trackHierarchyListener, null);
```
Comment (dgozman) [inline packages/playwright-core/src/server/reuseController.ts:65]: Ouch!

**Review by dgozman — APPROVED** [review]

File: packages/recorder/src/callLog.tsx (hunk 1 above)
Comment (pavelfeldman) [reply packages/recorder/src/callLog.tsx:57]: It is not stray, I needed to simplify the interface to remove channel dependencies.

File: packages/playwright-core/src/server/reuseController.ts (hunk 2 above)
Comment (pavelfeldman) [reply packages/playwright-core/src/server/reuseController.ts:65]: Ouch indeed!

**Pull request merged**